server/container_restore.go

```diff
@@ -75,6 +75,21 @@ func (s *Server) CRImportCheckpoint(
 
 	var restoreArchivePath string
 	if restoreStorageImageID != nil {
+		sb, err := s.getPodSandboxFromRequest(ctx, sbID) // Note that we might call getPodSandboxFromRequest with a different sbID later. Is that necessary?
```

Comment:
we haven't run
```
	if sbID == "" {
		// restore into previous sandbox
		sbID = dumpSpec.Annotations[annotations.SandboxID]
		ctrID = config.ID
	} else {
		ctrID = ""
	}

```
yet here (line 147) so i don't think we know for sure what's the sandbox ID yet. that's why I opted to have it below

Reply:
Yes, but _that_ sandbox ID comes from inside the metadata image.

For simplicity, suppose the node had a policy “deny anything from `docker.io`”; then any attempt to restore a snapshot using a metadata image from `docker.io` should fail.

I don’t _really_ know what I’m doing — I’m assuming Kubelet is always passing the same kind of metadata (at least, there’s only effectively a single `CreateContainer` call site in Kubelet), so using the same sandbox ID the non-snapshot-restore “create container” code path is using should be correct here.

But then again, I don’t know restoring a sandbox ID / namespace from contents of the metadata image follows namespace restrictions, so I might well be missing something.

Reply:
> suppose the node had a policy…

suppose the _namespace_ invoking the operation had a policy, rather.

Reply:
@adrianreber wdyt?

Reply:
The code to restore into an existing sandbox ID is a left-over from the initial pull request which supported to restore a pod. This is no longer possible with the current code. Most of the code was removed. This should haven been also removed.

Reply:
Further discussion to follow in https://github.com/cri-o/cri-o/pull/8706#discussion_r1813427411 .